spec/person_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rspec'
+require_relative '../classes/person'
+
+describe Person do
+  let(:person) { Person.new(25, 'John', parent_permission: false) }
+
+  it 'initializes with age, name, and parent permission' do
+    expect(person.age).to eq(25)
+    expect(person.name).to eq('John')
+    expect(person.instance_variable_get(:@parent_permission)).to be_falsey
+  end
+
+  it 'correctly returns the name' do
+    expect(person.correct_name).to eq('John')
+  end
+
+  it 'initializes with a unique ID' do
+    expect(person.instance_variable_get(:@id)).to be_a(Integer)
+  end
+  it 'can use services with parent permission' do
+    person = Person.new(15, 'Bob', parent_permission: true)
+    expect(person.can_use_services?).to be_truthy
+  end
+end
```

Comment:
- [ ] Good job so far but you have to add some tests for your person file, there are some scenarios that were missed 🙃 
  - The constructor method should return an empty rental as well but you just put name, age, a dn parent_permission,
  - You should test the scenario of an unknown name; if the name is not provided the default name should be `Unknown`. The constructor method sets a default value for the optional name parameter if it is not provided.
  - You should test the context when adding a new rental so the `add_rental` method returns a rental and adds it to the book.
  - You should test the context when checking permissions:
    - The `can_use_services?` method returns false if under age and no parent permission
    - The `can_use_services?` method returns true if over age
  - You should test the context when validating the name, the `correct_name` method corrects the name for example if the name is `alongnameiscorrected` it should be equal to `Alongnamei`